sdk/identity/identity/src/credentials/interactiveBrowserCredential.browser.ts

```diff
@@ -59,25 +64,28 @@ export class InteractiveBrowserCredential implements TokenCredential {
       auth: {
         clientId: options.clientId!, // we just initialized it above
         authority: `${options.authorityHost}/${options.tenantId}`,
-        knownAuthorities,
-        ...(options.redirectUri && { redirectUri: options.redirectUri }),
-        ...(options.postLogoutRedirectUri && { redirectUri: options.postLogoutRedirectUri })
+        knownAuthorities
       },
       cache: {
-        cacheLocation: "localStorage",
+        cacheLocation: "sessionStorage",
         storeAuthStateInCookie: true
       }
     };
 
-    this.msalObject = new msal.UserAgentApplication(this.msalConfig);
+    this.msalConfig.auth.redirectUri =
+      typeof options.redirectUri === "function" ? options.redirectUri() : options.redirectUri;
+    this.msalConfig.auth.postLogoutRedirectUri =
+      typeof options.postLogoutRedirectUri === "function"
+        ? options.postLogoutRedirectUri()
+        : options.postLogoutRedirectUri;
+
+    this.msalObject = new msalBrowser.PublicClientApplication(this.msalConfig);
   }
 
-  private login(): Promise<msal.AuthResponse> {
+  private async login(): Promise<msalBrowser.AuthenticationResult | null> {
```

Comment:
I'll remove the `async` here if it's unnecessary. I'm leaving this comment as as reminder.